recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
             return "libdispatch"
 
         if self.settings.os == "Windows":
             return "windows"
 
-        if self.settings.os == "Emscripten":
-            return "emscripten"
-
         return "portable"
 
     def _validate_task_system_libdispatch(self):
         if self.settings.os == "Linux":
             if self.settings.compiler != "clang":
-                raise ConanInvalidConfiguration("{}/{} task_system=libdispatch needs Clang compiler when using OS: {}. Use Clang compiler or switch to task_system=portable or task_system=auto".format(self.name, self.version, self.settings.os))
+                raise ConanInvalidConfiguration(f"{self.ref} task_system=libdispatch needs Clang compiler when using OS: {self.settings.os}. Use Clang compiler or switch to task_system=portable or task_system=auto")
         elif self.settings.os != "Macos":
-            raise ConanInvalidConfiguration("{}/{} task_system=libdispatch is not supported on {}. Try using task_system=auto".format(self.name, self.version, self.settings.os))
-
-    def _validate_task_system_windows(self):
-        if self.settings.os != "Windows":
-            self.output.info("Libdispatch is not supported on {}. The task system is changed to {}.".format(self.settings.os, self.options.task_system))
-            raise ConanInvalidConfiguration("{}/{} task_system=windows is not supported on {}. Try using task_system=auto".format(self.name, self.version, self.settings.os))
-
-    def _validate_task_system_emscripten(self):
-        if self.settings.os != "Emscripten":
-            raise ConanInvalidConfiguration("{}/{} task_system=emscripten is not supported on {}. Try using task_system=auto".format(self.name, self.version, self.settings.os))
+            raise ConanInvalidConfiguration(f"{self.ref} task_system=libdispatch is not supported on {self.settings.os}. Try using task_system=auto")
 
     def _validate_task_system(self):
         if self.options.task_system == "libdispatch":
             self._validate_task_system_libdispatch()
         elif self.options.task_system == "windows":
-            self._validate_task_system_windows()
-        elif self.options.task_system == "emscripten":
-            self._validate_task_system_emscripten()
+            if self.settings.os != "Windows":
+                raise ConanInvalidConfiguration(f"{self.ref} task_system=windows is not supported on {self.settings.os}. Try using task_system=auto")
+
+    def _default_thread_system(self):
+        if self.settings.os == "Macos":
+            return "pthread-apple"
+
+        if self.settings.os == "Linux":
+            return "pthread"
+
+        if self.settings.os == "Windows":
+            return "win32"
+
+        if self.settings.os == "Emscripten":
+            return "pthread-emscripten"
+
+        return "none"
+
+    def _validate_thread_system(self):
+        if self.options.thread_system == "pthread-apple":
+            if self.settings.os != "Macos":
```

Comment:
Could we maybe have less nesting?

```suggestion
        if self.options.thread_system == "pthread-apple" and self.settings.os != "Macos":
```